.github/workflows/tag-maven-deploy.yml

```diff
@@ -0,0 +1,75 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path
+
+name: tag push maven deploy
+
+on:
+  push:
+    tags:
+      - v[0-9].[0-9].[0-9]
+
+jobs:
+  publish-maven-central:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+
+    - name: Build with Maven
+      run: mvn -B package -DskipTests --file pom.xml
+
+    - name: Set up Apache Maven Central
+      uses: actions/setup-java@v3
+      with: # running setup-java again overwrites the settings.xml
+        java-version: '17'
+        distribution: 'temurin'
+        server-id: sonatype-nexus-repository # Value of the distributionManagement/repository/id field of the pom.xml
+        server-username: MAVEN_USERNAME # env variable for username in deploy
+        server-password: MAVEN_CENTRAL_TOKEN # env variable for token in deploy
+        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
+
+    - name: Publish to Apache Maven Central
+      run: mvn clean deploy -P maven-central-deploy -pl bifromq-plugin-auth-provider,bifromq-plugin-event-collector,bifromq-plugin-inbox-broker,bifromq-plugin-setting-provider -am -DskipTests
```

Comment:
remove bifromq-plugin-inbox-broker from deployment currently, it will be made public in later release

Reply:
OK, it will be removed in next commit.